packages/tre/src/runtime/index.ts

```diff
@@ -1,14 +1,36 @@
+import assert from "assert";
 import childProcess from "child_process";
 import crypto from "crypto";
 import fs from "fs";
 import os from "os";
 import path from "path";
+import workerdPath from "workerd";
 import { SERVICE_LOOPBACK, SOCKET_ENTRY } from "../plugins";
 import { Awaitable, MiniflareCoreError } from "../shared";
 
+let workerdVersion: string | undefined;
+// Gets the maximum supported compatibility date of this `workerd` version
+export function getWorkerdVersion(): string {
+  if (workerdVersion !== undefined) return workerdVersion;
+
+  const workerdRoot = path.resolve(path.dirname(workerdPath), "..");
+  const workerdPkgPath = path.join(workerdRoot, "package.json");
+  const workerdPkg = JSON.parse(fs.readFileSync(workerdPkgPath, "utf8"));
+
+  const version = workerdPkg.version.split(".");
+  assert(version.length >= 3); // <major>.<max_compatibility_date>.<patch>
+
+  const compatDate = version[1];
+  const length = compatDate.length;
+  assert(length >= 6); // YYYYMMDD
+  const year = compatDate.substring(0, length - 4);
+  const month = compatDate.substring(length - 4, length - 2);
+  const date = compatDate.substring(length - 2, length);
+  return (workerdVersion = `${year}-${month}-${date}`);
```

Comment:
I wonder if it would be better to run `workerd` here? `workerd --version` should output `workerd YYYY-MM-DD`, which might be a better way of getting this?

Reply:
`import path, { version, compatibilityDate } from 'workerd';`